semantics/src/main/kotlin/com/strumenta/kolasu/semantics/symbol/description/SymbolDescription.kt

```diff
@@ -3,46 +3,173 @@ package com.strumenta.kolasu.semantics.symbol.description
 import com.strumenta.kolasu.language.KolasuLanguage
 import com.strumenta.kolasu.model.Named
 import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.model.ReferenceByName
+import kotlin.reflect.KClass
+import kotlin.reflect.full.allSuperclasses
 
+/**
+ * The description of a [Node] from the outside,
+ * i.e. from other trees not containing the node.
+ * @property name the name of the symbol
+ * @property identifier the identifier of the described node
+ * @property type a description of the type of the described node
+ * @property fields the visible fields of the described node
+ *
+ * @author Lorenzo Addazi <lorenzo.addazi@strumenta.com>
+ **/
 data class SymbolDescription(
```

Comment:
I would not make SymbolDescription as a Node, because it is not part of an AST and Kolasu nodes represent only parts of the AST. Is it possible to do this change or would that have too much impact?

Reply:
We discussed and decided to define SymbolDescription as nodes some weeks ago and the internal implementations have been modified with this idea in mind. This could be avoided but it would have impacts on the Symbol Provider configuration API.

Reply:
Sorry, I do not remember why we decided to make SymbolDescriptions to extend Node. Do you remember the reason? Is that reason still true?

In order not to delay merging this PR I would then write down the rationale for making SymbolDescription extend Node, and add a note that this could be later reconsidered

Reply:
Initially, we thought it would have made integration with LionWeb easier - re-use logic for storing/loading Kolasu nodes. I don't know if that is still the case, but it would still be valid in case we would decide to store SymbolDescriptions inside a LionWeb repository. For the same reason, we defined the _Symbol Description Language_. Defining symbol descriptions as nodes is also useful as all traversal and other Kolasu APIs can be reused. Is there any document where we decided that Kolasu nodes only represent AST nodes?

Reply:
> Initially, we thought it would have made integration with LionWeb easier - re-use logic for storing/loading Kolasu nodes. I don't know if that is still the case,

I think this is not anymore the case. Sorry for the confusion. We can implement serialization to LionWeb (and deserialization) rather easily, without the need to force extending Node for that goal

> Defining symbol descriptions as nodes is also useful as all traversal and other Kolasu APIs can be reused

I see the point

> Is there any document where we decided that Kolasu nodes only represent AST nodes?

No, and there should be one.

I think that on a practical level is sometimes tempting to make something a Kolasu Node, to reuse the APIs we have. But on a conceptual level this causes confusion as we end up having Node indicating _mostly_ AST nodes and sometimes other things, for which the class Node have features that makes sense to use for proper AST nodes, but not for other Nodes.

Maybe in Kolasu 1.6 we could have two interfaces: one called GenericNode and one called ASTNode extending GenericNode, but I do not see clearly what abstraction would GenericNode represent